viewing_party/party.py

```diff
@@ -0,0 +1,160 @@
+    x = 1
+    user_watched_list = {watched['title']: watched['title'] for watched in user_data['watched']}
+    unique_data = [y for x in user_data['friends'] for y in x['watched'] if y.values() in user_watched_list.keys()]
+    return unique_data 
+
+def get_friends_unique_watched(user_data):
+    user_watched_list = {}
+    unique_data = []
+  
+    x = 1
+    for watched in user_data['watched']:
+        user_watched_list[watched['title']] = watched['title']
+
+    for each_friend in user_data['friends']:
+        if len(each_friend['watched']) > 0:
+            for x in each_friend['watched']:
+                for k, v in x.items():
+                    if v not in user_watched_list.keys():
+                        unique_data.append(v)
+           
+    set1 = set(unique_data)
+    list1 = list(set1)
+    final = []
+    for x in list1:
+        dict1 = {'title': x}
+        final.append(dict1)
+
+    return final  
+
+#wave 4 
+                      
+def get_available_recs(user_data):
+    user_dict_sub = {}
+    user_dict_watched = {}
+    recomendations = []
+
+    #creating dictionaires for user data
+    counter = 0
+    for subscription in user_data['subscriptions']:
+        user_dict_sub[counter] = subscription 
+        counter += 1
+    x = 0
+    for watched in user_data['watched']:
+        for k, v in watched.items():
+            user_dict_watched[x] = v
+            x += 1
+
+    #getting recomendations
+    for friend in user_data['friends']:
+        for watch in friend['watched']:
+            if watch['host'] in user_dict_sub.values() and watch['title'] not in user_dict_watched.values() and watch not in recomendations:
+                recomendations.append(watch)
+    return recomendations
+
+#Wave five
+def get_new_rec_by_genre(user_data):
+    genre_map = {}
+    if len(user_data['watched']) == 0:
+        return []
+    else:
+
+        for genre in user_data["watched"]:
+            if genre["genre"] not in genre_map:
+                genre_map[genre["genre"]] = 1 
+            else:
+                genre_map[genre["genre"]] += 1
+
+        max_key = max(genre_map, key=genre_map.get)
+        
+        users_watched_dict = {}
+        for title in user_data['watched']:
+            users_watched_dict[title['title']] = title['title']
+        
+        recomendations = []
+        for friends_watched in user_data['friends']:
+            for watch in friends_watched['watched']:
+                if watch['title'] not in users_watched_dict and watch['genre'] == max_key and watch not in recomendations:
+                    recomendations.append(watch)
+        return recomendations
+
+def get_rec_from_favorites(user_data):
+    if len(user_data['favorites']) == 0:
+        return []
+    else:
+        friend_watch_dict = {}
+        
+        for friend in user_data['friends']:
+            for friend_watch in friend['watched']:
+                friend_watch_dict[friend_watch['title']] = friend_watch['title']
```

Comment:
You seem to be making a dictionary that looks like this:

```python
{ 
  "game of thrones": "game of thrones",
  "wonder woman 1984": "wonder woman 1984",
}
```

Where each key-value pair is the same.  Maybe a set might be easier.